services/messaging/src/main/java/org/eclipse/hono/telemetry/impl/ForwardingTelemetryDownstreamAdapter.java

```diff
@@ -47,12 +47,7 @@ public ForwardingTelemetryDownstreamAdapter(final Vertx vertx, final SenderFacto
     }
 
     protected void forwardMessage(final ProtonSender sender, final Message msg, final ProtonDelivery upstreamDelivery) {
-        ProtonHelper.accepted(upstreamDelivery, true);
         sender.send(msg);
```

Comment:
IMHO we need to connect the downstream disposition handling with the upstream handling, i.e. analogous to the way it is down for events:
````
sender.send(msg, updatedDelivery -> upstreamDelivery.disposition(updatedDelivery.getRemoteState(), updatedDelivery.remotelySettled()));
````

Reply:
Since we stay now with AT_MOST_ONCE for the telemetry in my understanding it is the correct behavior to NOT send any disposition performative. The sender is not interested in any information since it is fire and forget. Also it is ca. 15% faster.

Reply:
Yes, now that we stay with AT_MOST_ONCE from adapters to Hono Messaging, the way it always was still is correct.

Reply:
Not exactly - before we have had the line:
```
  ProtonHelper.accepted(upstreamDelivery, true);
```
which creates disposition formative.

My proposal is to remove this.

Reply:
IMHO it is still relevant because the upstream receiver link's `autoAccept` flag is set to `false` in `UpstreamReceiverImpl` and we should (locally) settle the message in order to notify the proton-j libarary that we are done with the message and that it can be removed from the incoming list of messages awaiting settlement.

Reply:
I see - but then we should call 
'''
upstreamDelivery.settle();
'''
this "Causes the delivery to be removed from the connection's work list" and does not deliver a disposition frame to sender.

A disposition performatives should not be sendet with AT_MOST_ONCE.

WDYT?

Reply:
That's fine with me. However, I do not think that proton would send a disposition frame to the peer if we would stick with the existing code either, because it would (hopefully) see that the peer sends pre-settled messages (AT_MOST_ONCE). But anyway, go ahead with your changes. Maybe you can add a comment indicating why we are settling the message here since it has already led to discussions and might therefore be helpful once anybody re-visits the code ;-)

Reply:
After having taken a look at `org.apache.qpid.proton.engine.impl.DeliveryImpl` I actually think that we should expicitly do `ProtonHelper.accepted(upstreamDelivery, true);` instead of just `ProtonDelivery.settle()` because the `settle()` method does not update the local delivery state (ACCEPTED, in this case).

Reply:
In fact proton sends a disposition frame if we call ProtonHelper.accepted(upstreamDelivery, true); 

So the disposition of accepted is reflected in a disposition frame that will be sent to the sender. What is the effect of the local state change to accepted?

Reply:
> In fact proton sends a disposition frame if we call ProtonHelper.accepted(upstreamDelivery, true);

What makes you think so? Here is the code from `ProtonImpl.disposition()` which is called from `ProtonDeliveryImpl.disposition()`:

````
    public void disposition(final DeliveryState state)
    {
        _deliveryState = state;
        if(!_remoteSettled)
        {
            addToTransportWorkList();
        }
    }
````
So, if the message is pre-settled, I would assume that no disposition frame is sent.

There is no immediate effect of setting the local state except from the fact that it now reflects the state of the delivery accurately, on which other code may or may not rely on when invoking `ProtonDelivery.getLocalState()` ...

Reply:
I noticed this on the sender side - a disposition frame is sent. I just tried and saw that `upstreamDelivery.remotelySettled()` gives false, what in my understanding should be true. Let me dig into this..